c2corg_ui/views/document.py

```diff
@@ -432,12 +441,17 @@ def _redirect_to_full_url(self):
             raise HTTPInternalServerError(
                 "An error occurred while loading the document")
 
-        locale = data['locales'][0]
-        title = ''
-        if self._API_ROUTE == 'routes' and locale['title_prefix']:
-            title += locale['title_prefix'] + ' '
-        title += locale['title']
-        location = self.request.route_url(
-            self._API_ROUTE + '_view', id=data['document_id'],
-            lang=locale['lang'], slug=slugify(title))
-        raise HTTPMovedPermanently(location=location)
+        if 'redirects_to' in data:
+            if lang not in data['available_langs']:
+                lang = self._get_preferred_lang()
```

Comment:
What is "the best lang in data['available_langs']"?
On the UI side we don't know what lang to choose among ``available_langs`` if ``available_langs`` does not contain the requested ``lang``. In that case, I suggest to fall back on the user's perferred lang and let the API deal with it (return the "best lang" if the preferred lang is not available).
Doesn't it make sense?